### DISCUSSION TITLE
If Smart Meter Fails, the last known value keeps being displayed

### DISCUSSION THREAD

**Discussion by timaseth** [arendst/Tasmota#12484] [Ideas]
I have a tasmota node (NodeMCU) running the Energy Meter script and polling modbus data from an Eastron-like device.
The teleperiod is set to 10 (for MQTT reporting) and the script has Modbus poll set for 10 (x100; so 1 second).
Things work fine except...even with a power loss at the monitored circuit, the GUI keeps showing last valid data (which in fact, must start showing null values). Same for the MQTT reporting - no null values.
If, during such a power outage, the tasmota node is rebooted, I immediately get null readings.

Any pointers to rectify the situation?

BTW, Big Thanks for you work, devs!

**Comment by ascillato2** [comment]
Hi, Since you haven't any answer here, have you tried in the chat?

I don't have that device so I can not help you with that. Sorry.

**Reply by timaseth** [reply]
Thanks for reply, Adrian - I appreciate the care!
Discord chat is where I try first..always. This was plan B.

**Reply by barbudor** [reply]
@timaseth

> even with a power loss at the monitored circuit, the GUI keeps showing last valid data

Does it means that the meter is off too and do not reply to the modbus query ?
But Tasmota keep the last valid response and keep considering this is the actual power consumption ?

If I get it correctly, this is a difficult situation. I would expect that some would prefer that in case of loss of comm with the meter, it behave as it does today because it may be a temporary error and the circuit is still using power.
Maybe that would requires a setoption (if none already exist)

I don't deal much with Scripting and SML, but is there any possibility to deal with timeout error for example ?

**Reply by timaseth** [reply]
Hey @barbudor So nice to hear from you!
Si, Oui, you are right.
The Smart meter is monitiring mains and the tas node is on a UPS. When mains power is lost, the tas GUI/MQTT does not register null values (which..it should as the smart meter can no longer give modbus responces - ah..perhaps that is the issue, could it be? - lack of zeros being sent by the smart meter!).

So, tas takes the last known value and keeps displaying it (voltage, current and such when infact, there is none) and sending it via mqtt.

If during such a loss, I reboot the tas device, of course, null values are shown since there is no modbus responce.
As there are no so called errors anywhere, remote monitoring of the smart meter gives false readings.

Nothing extraordinary but I love tasmota way too much not to bring it to the attention of the devs - perhaps there is an easy fix.

Many thanks for your time!

**Reply by ascillato2** [reply]
Moved this from Q&A to Ideas, as it is now a Feature Request.
